client/landing/gutenboarding/components/domain-picker-button/index.tsx

```diff
@@ -44,89 +46,86 @@ const DomainPickerButton: React.FunctionComponent< Props > = ( {
 } ) => {
 	const buttonRef = React.createRef< HTMLButtonElement >();
 
-	const domainSuggestions = useDomainSuggestions( { locale: useI18n().i18nLocale, quantity: 10 } );
-
-	const domainCategories = useSelect( ( select ) =>
-		select( DOMAIN_SUGGESTIONS_STORE ).getCategories()
-	);
+	const [ domainSearch, setDomainSearch ] = React.useState< string >( '' );
 
-	const { domainSearch, domainCategory } = useSelect( ( select ) =>
-		select( STORE_KEY ).getState()
-	);
-	const { setDomainSearch, setDomainCategory } = useDispatch( STORE_KEY );
+	const [ domainPickerMode, setDomainPickerMode ] = React.useState<
+		'popover' | 'modal' | undefined
+	>();
 
-	const [ isDomainPopoverVisible, setDomainPopoverVisibility ] = React.useState( false );
-	const [ isDomainModalVisible, setDomainModalVisibility ] = React.useState( false );
-
-	const handlePopoverClose = ( e?: React.FocusEvent ) => {
-		// Don't collide with button toggling
-		if ( e?.relatedTarget === buttonRef.current ) {
-			return;
-		}
-		setDomainPopoverVisibility( false );
+	const handlePopoverToggle = () => {
+		setDomainPickerMode( ( mode ) => ( mode ? undefined : 'popover' ) );
 	};
 
-	const handleModalClose = () => {
-		setDomainModalVisibility( false );
+	const handleClose = () => {
+		setDomainPickerMode( undefined );
 	};
 
 	const handleMoreOptions = () => {
-		setDomainPopoverVisibility( false );
-		setDomainModalVisibility( true );
+		setDomainPickerMode( 'modal' );
 	};
 
+	const { __ } = useI18n();
+
+	const currentStep = useCurrentStep();
+
+	const { siteTitle, siteVertical } = useSelect( ( select ) => select( STORE_KEY ).getState() );
+	const currentUser = useSelect( ( select ) => select( USER_STORE ).getCurrentUser() );
+
+	const prioritisedSearch = domainSearch.trim() || siteTitle;
+	let searchVal;
+
+	if ( currentStep !== 'IntentGathering' ) {
+		searchVal =
+			prioritisedSearch ||
+			siteVertical?.label.trim() ||
+			currentUser?.username ||
+			__( 'My new site' );
+	} else {
+		searchVal = prioritisedSearch || '';
+	}
```

Comment:
Not a blocker for this PR but here is how I see the whole `domainSearch` handling:
- remove `domainSearch` local state from here.
- rename `domainSearch` prop to `initialDomainSearch` and use it to initialise `domainSearch` state inside DomainPicker component.
- get `initialDomainSearch` value from a hook in Gutenboarding (to be reused in Domain Step).
- remove `onSetDomainSearch` prop from DomainPicker.